packages/zoe/src/zoeService/zoe.js

```diff
@@ -0,0 +1,357 @@
+// @ts-check
+import makeIssuerKit from '@agoric/ertp';
+import makeWeakStore from '@agoric/weak-store';
+import { assert, details } from '@agoric/assert';
+import { E } from '@agoric/eventual-send';
+import { makePromiseKit } from '@agoric/promise-kit';
+
+/**
+ * Zoe uses ERTP, the Electronic Rights Transfer Protocol
+ */
+import '@agoric/ertp/exported';
+
+import '../../exported';
+import '../internal-types';
+
+import { makeIssuerTable } from '../issuerTable';
+import { makeZoeSeatAdminKit } from './zoeSeat';
+import zcfContractBundle from '../../bundles/bundle-contractFacet';
+import { arrayToObj } from '../objArrayConversion';
+import { cleanKeywords, cleanProposal } from './cleanProposal';
+import { makeHandle } from '../table';
+
+/**
+ * Create an instance of Zoe.
+ *
+ * @param {VatAdminSvc} vatAdminSvc - The vatAdmin Service, which carries the power
+ * to create a new vat.
+ * @returns {ZoeService} The created Zoe service.
+ */
+function makeZoe(vatAdminSvc) {
+  const invitationKit = makeIssuerKit('Zoe Invitation', 'set');
+
+  // Zoe state shared among functions
+  const issuerTable = makeIssuerTable();
+  /** @type {WeakSet<Installation>} */
+  const installations = new WeakSet();
+
+  /** @type {WeakStore<Instance,InstanceAdmin>} */
+  const instanceToInstanceAdmin = makeWeakStore('instance');
+
+  /** @type {GetAmountMath<any>} */
+  const getAmountMath = brand => issuerTable.get(brand).amountMath;
+
+  /** @type {WeakStore<Brand, ERef<Purse>>} */
+  const brandToPurse = makeWeakStore('brand');
+
+  /**
+   * Create an installation by permanently storing the bundle. It will be
+   * evaluated each time it is used to make a new instance of a contract.
+   */
+  /** @type {Install} */
+  const install = async bundle => {
+    /** @type {Installation} */
+    const installation = { getBundle: () => bundle };
+    harden(installation);
+    installations.add(installation);
+    return installation;
+  };
+
+  /** @type {ZoeService} */
+  const zoeService = {
+    getInvitationIssuer: () => invitationKit.issuer,
+    install,
+    getPublicFacet: instance =>
+      instanceToInstanceAdmin.get(instance).getPublicFacet(),
+    getBrands: instance => instanceToInstanceAdmin.get(instance).getBrands(),
+    getIssuers: instance => instanceToInstanceAdmin.get(instance).getIssuers(),
+    getTerms: instance => instanceToInstanceAdmin.get(instance).getTerms(),
+    startInstance: async (
+      installation,
+      uncleanIssuerKeywordRecord = harden({}),
+      terms = harden({}),
+    ) => {
+      // Unpack the invitationKit.
+
+      const {
+        issuer: invitationIssuer,
+        mint: invitationMint,
+        amountMath: invitationAmountMath,
+      } = invitationKit;
```

Comment:
None of this is used until line 220. Let's move it down much lower.   

after `instanceToInstanceAdmin.init(instance, instanceAdmin);` seems like a good place.